File: src-tauri/src/domain/note.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "snake_case")]
pub enum NoteType {
    Text,
    Todo,
    Expense,
    Photo,
}

impl NoteType {
    pub fn as_str(&self) -> &'static str {
        match self {
            Self::Text => "text",
            Self::Todo => "todo",
            Self::Expense => "expense",
            Self::Photo => "photo",
        }
    }
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct NoteSummary {
    pub id: i64,
    pub note_type: String,
    pub title: String,
    pub color: String,
    pub x: f64,
    pub y: f64,
    pub width: f64,
    pub height: f64,
    pub folder_id: Option<i64>,
    pub open: bool,
    pub is_deleted: bool,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct TextNote {
    pub note: NoteSummary,
    pub content: String,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateNoteInput {
    pub note_type: NoteType,
    pub title: String,
    pub color: Option<String>,
    pub folder_id: Option<i64>,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateTextNoteInput {
    pub note_id: i64,
    pub content: String,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateNoteColorInput {
    pub note_id: i64,
    pub color: String,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SetDeletedNoteInput {
    pub note_id: i64,
    pub is_deleted: bool,
}

File: src/types/note.ts
export type NoteType = "text" | "richText" | "todo" | "expense" | "photo";

export interface NoteSummary {
  id: number;
  noteType: NoteType;
  title: string;
  color: string;
  x: number;
  y: number;
  width: number;
  height: number;
  folderId: number | null;
  open: boolean;
  isDeleted: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface TextNote {
  note: NoteSummary;
  content: string;
}

export interface TodoItem {
  id: number;
  noteId: number;
  content: string;
  completed: boolean;
  position: number;
}

export interface TodoNote {
  note: NoteSummary;
  items: TodoItem[];
}

export interface ExpenseItem {
  id: number;
  noteId: number;
  description: string;
  amount: number;
  kind: "수입" | "지출";
  category: string;
  expenseDate: string;
  position: number;
}

export interface ExpenseNote {
  note: NoteSummary;
  items: ExpenseItem[];
  total: number;
}

export interface PhotoItem {
  id: number;
  noteId: number;
  filePath: string;
  position: number;
}

export interface PhotoNote {
  note: NoteSummary;
  items: PhotoItem[];
}

export interface FolderType {
  id: number;
  name: string;
  parent_id: number;
  sort_order: number;
  created_at: Date;
  updated_at: Date;
}

export interface SelectOption {
  value: string;
  label: string;
  disabled?: boolean;
}